web/app/help/operations/find-optimal-spanning-tree.asciidoc

```diff
@@ -0,0 +1,28 @@
+### Find optimal spanning tree
+
+Finds the https://en.wikipedia.org/wiki/Minimum_spanning_tree[minimum (or maximum) spanning tree]
+in a graph. The edges marked by the emitted edge attribute (``in_tree`` by default)
+form a tree for each component in the graph. This tree will have the lowest
+(or highest) possible total edge weight.
+
+Uses the
+https://networkit.github.io/dev-docs/cpp_api/classNetworKit_1_1RandomMaximumSpanningForest.html[NetworKit]
+implementation.
+
+====
+[p-name]#Save as#::
+The new edge attribute will be created under this name. Edges that make up the
+tree are marked by 1 and the edges that are not part of the tree are marked by 0.
```

Comment:
I wonder if 1 and undefined would be better? It would be less space and I think we do stuff like that in other operations. Maybe less convenient, though? Not sure.

Reply:
> I wonder if 1 and undefined would be better? It would be less space and I think we do stuff like that in other operations. Maybe less convenient, though? Not sure.

I wasn't sure either. I don't see much precedent. "Add reversed edges" uses 0 and 1.

For visualization undefined is a bit worse because it will be the default blue and the color assigned to 1 will not contrast with it in many color scales.

For filtering they are both fine.

For derived attributes and Python use... I'm not sure.

But you're right about the storage on the Spark side! I've sent a PR to switch to undefined/1: #118. I'll leave it for you to decide whether to merge it.

Reply:
(To keep the discussion in one place let me briefly answer here.)

I'm torn because of the visualization thing and also if you look at things with operations research eyes then 0 is more logical, but still, I'll go ahead and merge that PR. Mostly for consistency. Otherwise I wouldn't have a good way to explain why the find Steiner tree op works differently. And also we use this undefined convention in inputs as well, e.g. shortest paths and well, Steiner tree again.